util/flights/flights.go

```
@@ -0,0 +1,192 @@
+	data, err := doQuery(flight, date)
+	if err != nil {
+		return nil, fmt.Errorf("querying for flight %q: %w", flight, err)
+	}
+
+	return data, nil
+}
+
+func doQuery(flight, date string) (*Data, error) {
+	url := fmt.Sprintf("%s/%s/%s", apiBase, strings.ReplaceAll(flight, " ", ""), date)
+
+	req, err := http.NewRequest("GET", url, nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)